optimum/onnxruntime/optimization.py

```diff
@@ -0,0 +1,274 @@
+                they exist.
+            resume_download (:obj:`bool`, `optional`, defaults to :obj:`False`):
+                Whether or not to delete incompletely received file. Attempts to resume the download if such a file
+                exists.
+            revision(:obj:`str`, `optional`):
+                The specific version to use. It can be a branch name, a tag name, or a commit id, since we use a
+                git-based system for storing models and other artifacts on huggingface.co, so ``revision`` can be any
+                identifier allowed by git.
+        """
+        config_kwargs_default = [
+            ("cache_dir", None),
+            ("force_download", False),
+            ("resume_download", False),
+            ("revision", None),
+        ]
+        config_kwargs = {name: kwargs.get(name, default_value) for (name, default_value) in config_kwargs_default}
+        self.cache_dir = config_kwargs.get("cache_dir")
+        self.model_name_or_path = model_name_or_path
+        if not isinstance(ort_config, ORTConfig):
+            ort_config = ORTConfig.from_pretrained(ort_config, **config_kwargs)
+        self.ort_config = ort_config
+        self.onnx_config = None
+        self.feature = feature
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path)
+        self.model = FeaturesManager.get_model_from_feature(self.feature, self.model_name_or_path)
+        self.onnx_model_path = None
+        self.optim_model_path = None
+
+    def export(self, model_path: os.PathLike) -> None:
+        """
+        Load and export a model to an ONNX Intermediate Representation (IR).
+
+        Param:
+            model_path (:obj:`os.PathLike`):
+                The path used to save the model exported to an ONNX Intermediate Representation (IR).
+        """
+
+        model_type, model_onnx_config = FeaturesManager.check_supported_model_or_raise(
+            self.model, feature=self.feature
+        )
+        self.onnx_config = model_onnx_config(self.model.config)
+        opset = self.onnx_config.default_onnx_opset if self.ort_config.opset is None else self.ort_config.opset
+        _ = export(self.tokenizer, self.model, self.onnx_config, opset, model_path)
+
+    def fit(self, output_dir: Union[str, os.PathLike], **kwargs) -> None:
+        """
+        Load and export a model to an ONNX Intermediate Representation (IR) and apply the graph optimization.
+
+        Param:
+            output_dir (:obj:`Union[str, os.PathLike]`):
+                The output directory where the optimized model will be saved.
+        """
+        output_dir = output_dir if isinstance(output_dir, Path) else Path(output_dir)
+        self.onnx_model_path = output_dir.joinpath("model.onnx")
+        self.optim_model_path = generate_identified_filename(self.onnx_model_path, "-optimized")
+
+        self.export(self.onnx_model_path)
+        config = self.onnx_config._config
```

Comment:
We could use `self.model.config` instead of `self.onnx_config._config`

Reply:
✅